src/combinatorics/catalan.lean

```diff
@@ -129,3 +144,48 @@ by norm_num [catalan_eq_central_binom_div, nat.central_binom, nat.choose]
 
 lemma catalan_three : catalan 3 = 5 :=
 by norm_num [catalan_eq_central_binom_div, nat.central_binom, nat.choose]
+
+namespace tree
+open_locale tree
+
+/-- Given two finsets, find all trees that can be formed with
+  left child in `a` and right child in `b` -/
+@[simp] def pairwise_node (a : finset (tree unit)) (b : finset (tree unit)) : finset (tree unit) :=
+  (a ×ˢ b).map ⟨λ x, x.1 △ x.2, λ ⟨x₁, x₂⟩ ⟨y₁, y₂⟩, by { simp, tauto, }⟩
+
+/-- A finset of all trees with `n` nodes. See `mem_trees_of_nodes_eq` -/
+def trees_of_num_nodes_eq : ℕ → finset (tree unit)
```

Comment:
This name is a bit longish for what it does, and it contains many other name atoms. Can you improve it?

Reply:
I tried to think of some other names myself, but nothing came to mind. I think we should pretend that the definition has the following form
```lean
def trees_of_num_nodes_eq (n : ℕ) : finset (tree unit) :=
set.to_finset {x : tree unit | x.num_nodes = n}
```
and name it accordingly.

Reply:
This was my thought process as well -- we cannot directly define `set.to_finset {x | x.num_nodes = n}` because then we would need a proof of finiteness, which is exactly what we are going to show (and for those who care, it is nonconstructive). But we pretend that this is the definition, and that was how the naming decision was made.

I agree that I'm still unsure of the name, however, and I'm open to suggestions for a better one.

Reply:
Can we add a proof that this finset, coerced to a set, is equal to what Kyle writes? It should be easy with `mem_trees_of_nodes_eq`